sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -174,6 +174,7 @@ object CatalystTypeConverters {
             convertedIterable += elementConverter.toCatalyst(item)
           }
           new GenericArrayData(convertedIterable.toArray)
+        case g: GenericArrayData => new GenericArrayData(g.array.map(elementConverter.toCatalyst))
```

Comment:
`toCatalystImpl` converts Scala data into Catalyst one but  `GenericArrayData` is Catalyst-internal, so this change looks weried.

Reply:
> `toCatalystImpl` converts Scala data into Catalyst one but `GenericArrayData` is Catalyst-internal, so this change looks weried.

For complex type, after use JsonToStruct, don't need this converter any more. So I remove this.
How about current change?